cosmic-update.jenkinsfile

```diff
@@ -0,0 +1,147 @@
+import groovy.json.JsonSlurper
+def currentRelease
+def previousRelease
+pipeline
+{
+	agent any
+	stages
+	{
+		// This stage checks that an upstream step, UniProt Update, was run successfully.
+		stage('Check UniProt Update build succeeded')
+		{
+			steps
+			{
+				script
+				{
+					// Get current release number from directory
+					currentRelease = (pwd() =~ /Releases\/(\d+)\//)[0][1];
+					previousRelease = (pwd() =~ /Releases\/(\d+)\//)[0][1].toInteger() - 1;
+					// This queries the Jenkins API to confirm that the most recent build of 'UniProt Update' was successful.
+					def uniprotStatusUrl = httpRequest authentication: 'jenkinsKey', validResponseCodes: "${env.VALID_RESPONSE_CODES}", url: "${env.JENKINS_JOB_URL}/job/${currentRelease}/job/Pre-Slice/job/UniProtUpdate/lastBuild/api/json"
+					if (uniprotStatusUrl.getStatus() == 404)
+					{
+						error("UniProt Update has not yet been run. Please complete a successful build.")
+					}
+					else
+					{
+						def uniprotStatusJson = new JsonSlurper().parseText(uniprotStatusUrl.getContent())
+						if (uniprotStatusJson['result'] != "SUCCESS")
+						{
+							error("Most recent UniProt Update build status: " + uniprotStatusJson['result'] + ". Please complete a successful build.")
+						}
+					}
```

Comment:
This whole block can be replaced by the shared library's `checkUpstreamBuildSucceeded("UniProtUpdate")` method

Reply:
I've update the Jenkinsfile to make use of the shared library.